src/Unleash/DefaultUnleash.cs

```diff
@@ -124,14 +126,31 @@ private bool CheckIsEnabled(string toggleName, UnleashContext context, bool defa
             }
             else
             {
-                enabled = featureToggle.Strategies.Any(s => GetStrategyOrUnknown(s.Name).IsEnabled(s.Parameters, enhancedContext, ResolveConstraints(s).Union(s.Constraints)));
+                strategy = featureToggle.Strategies
+                    .FirstOrDefault(s =>
+                        GetStrategyOrUnknown(s.Name)
+                        .IsEnabled(s.Parameters, enhancedContext, ResolveConstraints(s).Union(s.Constraints))
+                    );
+                enabled = strategy != null;
+            }
+
+            if (strategy != null)
+            {
+                string groupId = null;
+                strategy.Parameters.TryGetValue("groupId", out groupId);
+                variant = enabled ? VariantUtils.SelectVariant(groupId, context, strategy.Variants) : null;
+            }
+
+            if (variant == null && defaultVariant != null)
+            {
+                variant = enabled ? VariantUtils.SelectVariant(featureToggle, context, defaultVariant) : defaultVariant;
```

Comment:
We are doing same enabled check in two if statements. Can we just pull it out and do 

if( !enabled)
variant = defaultVariant
else { 
doOtherStuff